src/main/java/io/stargate/sgv2/jsonapi/api/model/command/deserializers/SortClauseDeserializer.java

```diff
@@ -51,6 +52,25 @@ public SortClause deserialize(JsonParser parser, DeserializationContext ctxt) th
     int totalFields = sortNode.size();
     List<SortExpression> sortExpressions = new ArrayList<>(sortNode.size());
 
+    // $lexical is only allowed alone: handle first
+    JsonNode lexicalValue = sortNode.get(DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
+    if (lexicalValue != null) {
+      if (sortNode.size() > 1) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
+            "if sorting by '%s' no other sort expressions allowed",
+            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
+      }
+      if (!lexicalValue.isTextual()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
+            "if sorting by '%s' value must be STRING, not %s",
+            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD, lexicalValue.getNodeType());
+      }
+      // We cannot yet determine if lexical sort supported by the collection, just
```

Comment:
As per earlier note, deserializer validates that Sort expression seems ok, but cannot check if Lexical sort actually available on DB at this point.